src/main/java/org/opentripplanner/routing/edgetype/TripPattern.java

```diff
@@ -127,16 +124,27 @@
     /** Used by the MapBuilder (and should be exposed by the Index API). */
     public LineString geometry = null;
 
+
+    // TODO This should probably be precalculated in the PatternHopFactory
+    public LineString getHopGeometry(int stopIndex) {
+        if (hopGeometries != null) {
+            return hopGeometries[stopIndex];
+        } else {
+            Coordinate c1 = new Coordinate(stopPattern.stops[stopIndex].getLon(),
+                                            stopPattern.stops[stopIndex].getLat());
+            Coordinate c2 = new Coordinate(stopPattern.stops[stopIndex + 1].getLon(),
+                                            stopPattern.stops[stopIndex + 1].getLat());
+
+            return GeometryUtils.getGeometryFactory().createLineString(new Coordinate[] { c1, c2 });
+        }
+    }
+
```

Comment:
This method appear in the _field_ declaration section, I will move it to the end of the file.